crates/ruff_linter/src/rules/flake8_simplify/rules/split_of_static_string.rs

```diff
@@ -0,0 +1,175 @@
+    Expr::List(ExprList {
+        elts: list_items
+            .iter()
+            .map(|list_item| {
+                Expr::StringLiteral(ExprStringLiteral {
+                    value: StringLiteralValue::single(StringLiteral {
+                        value: (*list_item).to_string().into_boxed_str(),
+                        range: TextRange::default(),
+                        flags: StringLiteralFlags::default(),
+                    }),
+                    range: TextRange::default(),
+                })
+            })
+            .collect(),
+        ctx: ExprContext::Load,
+        range: TextRange::default(),
+    })
+}
+
+fn split_default(str_value: &str, max_split: usize) -> Option<Expr> {
+    // From the Python documentation:
+    // > If sep is not specified or is None, a different splitting algorithm is applied: runs of
+    // > consecutive whitespace are regarded as a single separator, and the result will contain
+    // > no empty strings at the start or end if the string has leading or trailing whitespace.
+    // > Consequently, splitting an empty string or a string consisting of just whitespace with
+    // > a None separator returns [].
+    // https://docs.python.org/3/library/stdtypes.html#str.split
+    if max_split == 0 {
+        let list_items: Vec<&str> = str_value.split_whitespace().collect();
+        Some(construct_replacement(&list_items))
+    } else {
+        // Autofix for maxsplit without separator not yet implemented
+        // split_whitespace().remainder() is still experimental:
+        // https://doc.rust-lang.org/std/str/struct.SplitWhitespace.html#method.remainder
+        None
+    }
+}
+
+fn split_sep(str_value: &str, sep_value: &str, max_split: usize, direction_left: bool) -> Expr {
+    let list_items: Vec<&str> = if direction_left && max_split > 0 {
+        str_value.splitn(max_split + 1, sep_value).collect()
+    } else if !direction_left && max_split > 0 {
+        str_value.rsplitn(max_split + 1, sep_value).collect()
+    } else if direction_left && max_split == 0 {
+        str_value.split(sep_value).collect()
+    } else {
+        str_value.rsplit(sep_value).collect()
+    };
+    construct_replacement(&list_items)
+}
+
+/// RUF035
+pub(crate) fn split_of_static_string(
+    checker: &mut Checker,
+    attr: &str,
+    call: &ExprCall,
+    str_value: &str,
+) {
+    let ExprCall { arguments, .. } = call;
+
+    let sep_arg = arguments.find_argument("sep", 0);
+    let maxsplit_arg = arguments.find_argument("maxsplit", 1);
+
+    // `split` vs `rsplit`
+    let direction_left = attr == "split";
+
+    let maxsplit_value = if let Some(maxsplit) = maxsplit_arg {
+        match maxsplit {
+            // Python allows maxsplit to be set to -1
+            Expr::UnaryOp(ExprUnaryOp {
+                op: UnaryOp::USub,
+                operand,
+                ..
+            }) if matches!(**operand, Expr::NumberLiteral { .. }) => 0,
```

Comment:
I think we should bail here if `maxplit` is a negative number other than `-1`

Reply:
It's rare, but `maxsplit=-2` behaves as if maxsplit is omitted.
From that perspective, any negative number could be flagged as non-idiomatic default, but that would be a different rule.

Working code in the wild:

https://github.com/localstack/localstack/blob/master/localstack-core/localstack/services/kms/utils.py#L17